### PR TITLE
Logging: Add support for resource requests

### PR DESCRIPTION
**Pull request by madddi**
This adds support for resource request configuration in openshift_logging  (#4320).

openshift_logging_elasticsearch seems to be in a state of migration / refactoring so the current and new variables don't match up with the parameters in openshift_logging. Is there anything I can or should do?

### CODE REVIEW

File: roles/openshift_logging/tasks/install_logging.yaml [hunk 1]
```diff
@@ -176,11 +176,15 @@
...
     openshift_logging_kibana_memory_limit: "{{ openshift_logging_kibana_ops_memory_limit }}"
+    openshift_logging_kibana_cpu_request: "{{ openshift_logging_kibana_ops_cpu_request }}"
+    openshift_logging_kibana_memory_request: "{{ openshift_logging_kibana_ops_memory_request }}"
     openshift_logging_kibana_hostname: "{{ openshift_logging_kibana_ops_hostname }}"
     openshift_logging_kibana_replicas: "{{ openshift_logging_kibana_ops_replica_count }}"
     openshift_logging_kibana_proxy_debug: "{{ openshift_logging_kibana_ops_proxy_debug }}"
     openshift_logging_kibana_proxy_cpu_limit: "{{ openshift_logging_kibana_ops_proxy_cpu_limit }}"
     openshift_logging_kibana_proxy_memory_limit: "{{ openshift_logging_kibana_ops_proxy_memory_limit }}"
+    openshift_logging_kibana_proxy_cpu_request: "{{ openshift_logging_kibana_ops_proxy_cpu_request }}"
+    openshift_logging_kibana_proxy_memory_request: "{{ openshift_logging_kibana_ops_proxy_memory_request }}"
```
Comment (ewolinetz) [inline roles/openshift_logging/tasks/install_logging.yaml]: we're also missing this for the ES and Fluentd `include_role`s

**Comment by ewolinetz** [comment]
bot, test pull request

**Comment by ewolinetz** [comment]
[test_logging]

File: roles/openshift_logging/tasks/install_logging.yaml (hunk 1 above)
Comment (madddi) [inline roles/openshift_logging/tasks/install_logging.yaml]: Yes, as I mentioned in my comment above. The variables in openshift_logging_elasticsearch have been renamed and removed from the include_role tasks. I didn't if someone is currently working on this. I could add the necessary parameters again, if you want me to.

**Comment by ewolinetz** [comment]
[test_logging]

**Comment by sdodson** [comment]
[test_logging]

File: roles/openshift_logging_curator/templates/curator.j2 [hunk 2]
```diff
@@ -45,6 +45,11 @@ spec:
 {% if curator_memory_limit is defined and curator_memory_limit is not none and curator_memory_limit != "" %}
               memory: "{{curator_memory_limit}}"
 {% endif %}
+            requests:
+              cpu: "{{curator_cpu_request}}"
+{% if curator_memory_request is defined and curator_memory_request is not none and curator_memory_request != "" %}
+              memory: "{{curator_memory_request}}"
+{% endif %}
```
Comment (richm) [inline roles/openshift_logging_curator/templates/curator.j2]: Why is only `memory` set conditionally?  Why is `cpu` always set but not `memory`?

File: roles/openshift_logging_kibana/templates/kibana.j2 [hunk 3]
```diff
@@ -47,6 +47,13 @@ spec:
 {% if kibana_memory_limit is not none and kibana_memory_limit != "" %}
               memory: "{{ kibana_memory_limit }}"
 {% endif %}
+            requests:
+{% if kibana_cpu_request is not none and kibana_cpu_request != "" %}
+              cpu: "{{ kibana_cpu_request }}"
+{% endif %}
+{% if kibana_memory_request is not none and kibana_memory_request != "" %}
+              memory: "{{ kibana_memory_request }}"
+{% endif %}
```
Comment (richm) [inline roles/openshift_logging_kibana/templates/kibana.j2]: Is it possible to have `requests` with no values in it?  Will this cause problems?

File: roles/openshift_logging_mux/templates/mux.j2 [hunk 4]
```diff
@@ -46,6 +46,13 @@ spec:
 {% if mux_memory_limit is not none %}
             memory: "{{mux_memory_limit}}"
 {% endif %}
+          requests:
+{% if mux_cpu_request is not none %}
+            cpu: "{{mux_cpu_request}}"
+{% endif %}
+{% if mux_memory_request is not none %}
+            memory: "{{mux_memory_request}}"
+{% endif %}
```
Comment (richm) [inline roles/openshift_logging_mux/templates/mux.j2]: Is it possible that the `requests` section can have no values?  Is the `not none` test sufficient to exclude the empty string value `''`?

File: roles/openshift_logging_elasticsearch/templates/es.j2 [hunk 5]
```diff
@@ -48,7 +48,10 @@ spec:
               cpu: "{{es_cpu_limit}}"
 {% endif %}
             requests:
-              memory: "512Mi"
+              memory: "{{es_memory_request}}"
+{% if es_cpu_request is defined and es_cpu_request is not none %}
+              cpu: "{{es_cpu_request}}"
+{% endif %}
```
Comment (richm) [inline roles/openshift_logging_elasticsearch/templates/es.j2]: Why is `cpu` conditionally defined but not `memory`?

**Comment by madddi** [comment]
@richm An empty `request` section is not valid, I think. I'm going to fix the checks there, thanks.

I just extended the tests that were already there. I'm not quite sure why `cpu` is conditionally defined and `memory` is not. It would be more consistent to conditionally set both values I think. I could change this behaviour as well if preferred.

**Comment by richm** [comment]
> I just extended the tests that were already there. I'm not quite sure why cpu is conditionally defined and memory is not. It would be more consistent to conditionally set both values I think. I could change this behaviour as well if preferred.

Yes, please.

**Comment by michaelgugino** [comment]
Closing due to inactivity.

**Pull request closed without merging**